pydoclint/utils/return_yield_raise.py

```diff
@@ -27,6 +27,19 @@ def _isNone(node: ast.AST) -> bool:
     return isinstance(node, ast.Constant) and node.value is None
 
 
+def isReturnAnnotationNoReturn(node: FuncOrAsyncFuncDef) -> bool:
+    """Check whether the return type annotation if `NoReturn`"""
+    if node.returns is None:
+        return False
+
+    returnAnnotation: str = unparseAnnotation(node.returns)
+    return returnAnnotation.startswith('NoReturn')
+
+
+def _isNoReturn(node: ast.AST) -> bool:
```

Comment:
This function doesn't seem to be used anywhere?

Reply:
And comparing the implementation of this function and that of `isReturnAnnotationNoReturn()`, it seems that the implementation of this function is better?

Reply:
Hi @jsh9!

> This function doesn't seem to be used anywhere?

Yeah, good catch. This is a leftover from a version that didn't work. I should (maybe?) remove it.

> And comparing the implementation of this function and that of `isReturnAnnotationNoReturn()`, it seems that the implementation of this function is better?

It looks better but it didn't work for me. I'm not familiar with the `ast` stuff here, so not sure if there is a way to make it work, but when I tried it it seems like `NoReturn` is not a `ast.Constant` but a `ast.Name` IIRC., this is why I had to go with the string comparison. I think I could also just do `returnAnnotation == 'NoReturn'`, the `startwidth()` was based on the generator stuff but in this case I think it should only match to `NoReturn`.